### PR TITLE
Arrive iOS Sample sdk update

### PR DESCRIPTION
**Pull request by naylin27**
update iOS sample app to latest SDK

### CODE REVIEW

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist [hunk 1]
```diff
@@ -13,23 +13,45 @@
...
-	<string>APPL</string>
+	<string>$(PRODUCT_BUNDLE_PACKAGE_TYPE)</string>
 	<key>CFBundleShortVersionString</key>
 	<string>1.0</string>
 	<key>CFBundleVersion</key>
 	<string>1</string>
 	<key>LSRequiresIPhoneOS</key>
 	<true/>
+	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
+	<string>Program requires GPS to track location </string>
```
Comment (amitmats) [inline iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:24]: Keep `NSLocationAlwaysAndWhenInUseUsageDescription`, `NSLocationAlwaysUsageDescription` and `NSLocationWhenInUseUsageDescription` to
   
`The app uses your location to help the store prepare for your arrival when you have an open order.`

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 2]
```diff
@@ -24,27 +24,29 @@ class ViewController: UIViewController, CSUserSessionDelegate {
     @IBOutlet var statusLabel: UILabel!
     @IBOutlet var trackingIdentifierButton: UIButton!
     @IBOutlet var trackingButton: UIButton!
-
     override func viewDidLoad() {
+
```
Comment (amitmats) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:28]: It would be good to show `NSLocationTemporaryUsageDescriptionDictionary` example  in sample app and request for temporary access to a user's precise location data ( in iOS 14 if Reduced). 

```swift
func locationManagerDidChangeAuthorization(_ manager: CLLocationManager) {
    let precise = manager.accuracyAuthorization

    switch precise {
    // Default Precision Level
    case .fullAccuracy:
        break
    // User Reduced Manually
    case .reducedAccuracy:
        // Create Temporary Request for Precise Location
        manager.requestTemporaryFullAccuracyAuthorization(withPurposeKey: "Curbside")
        break
    default:
        break
    }
}
```

As part of that code, you'll notice there is a request for temporary access to a user's precise location data. This function is extremely valuable for instances when the user has turned the Precise: Off, however your app still needs accurate location information in order to facilitate a better user experience. This request, when granted, will provide Precise location data for that user for the rest of the app's session; once the app is terminated, either by the user or OS, this Precise location data will no longer be available. This request can be made more than once.

you will need to include a specific usage string in your app's info.plist. This will explain to the user why your app is requesting temporary access to the Precise location data, and helps provide a better user experience. The following is an example of a reason that might be included as part of a temporary request to the user's Precise location data:

```
<key>NSLocationTemporaryUsageDescriptionDictionary</key>
<dict>
 <key>Curbside</key>
 <string>Your precise location will be used to help the store get your order to you as quickly as possible when you arrive.</string>
</dict>
```

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist [hunk 3]
```diff
@@ -13,23 +13,45 @@
...
 	<key>NSLocationAlwaysUsageDescription</key>
-	<string>UserSessionApp needs Always authorization</string>
-    <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
-    <string>UserSessionApp needs Always authorization</string>
-    <key>NSLocationWhenInUseUsageDescription</key>
-    <string>UserSessionApp needs Always authorization</string>
+	<string>Program requires GPS to track location </string>
+	<key>NSLocationWhenInUseUsageDescription</key>
+	<string>Program requires GPS to track location </string>
```
Comment (amitmats) [inline iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:28]: Add `NSLocationTemporaryUsageDescriptionDictionary`

**Comment by radwar13** [comment]
1) what are the reasons driving this change? Yes we need a new flow for the streamlined one, but this is not it. 

2) The code below will not work on iOS 13, it will fail to compile.

**Comment by amitmats** [comment]
The current sample apps on the is on old very version and does not compile. Idea here is keep it minimal/simple as it was before, not the streamlined one.

**Comment by radwar13** [comment]
Then fix for the compile issue and not add the temporary usage path. 
This needs to be changed once we get the simplified flow.

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 4]
```diff
@@ -24,27 +24,29 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
-        NotificationCenter.default.removeObserver(self)
-    }
+            super.viewWillAppear(animated)
+            NotificationCenter.default.removeObserver(self)
+        }
     
     @objc private func sessionValidated(notification: Notification) {
-        if let trackingIdentifier = CSUserSession.current().trackingIdentifier, !(trackingIdentifier.characters.isEmpty) {
+        if let trackingIdentifier = CSUserSession.current().trackingIdentifier, !(trackingIdentifier.isEmpty) {
+            tidTextField.text = trackingIdentifier
```
Comment (MathStPierre) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:49]: `tidTextField.text = trackingIdentifier` is appearing twice, one should be removed 😉

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 5]
```diff
@@ -73,49 +75,49 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
     private func updateButtons() {
-        trackingIdentifierButton.setTitle(hasTrackingID ? "Unregister Tracking Identfier" : "Set Tracking Identifier", for: .normal)
-        trackingButton.setTitle(isStartTrack ? "Start Tracking Site" : "Stop Tracking Site", for: .normal)
-    }
-
+          trackingIdentifierButton.setTitle(hasTrackingID ? "Unregister Tracking Identfier" : "Set Tracking Identifier", for: .normal)
+          trackingButton.setTitle(isStartTrack ? "Start Tracking Site" : "Stop Tracking Site", for: .normal)
+      }
+    
     @IBAction func startTracking(_ sender: Any) {
```
Comment (MathStPierre) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:82]: This method seems to toggle tracking from start to cancel. Maybe we could rename the method `startCancelTracking` or `toggleTracking` 🤔

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 6]
```diff
@@ -73,49 +75,49 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
-                errorTitleLabel.text = "Empty Tracking Identifier.";
-                return;
-            }
-            
-            if (trackToken.characters.count == 0) {
-                errorTitleLabel.text = "Empty track token.";
-                return;
-            }
+            resetError()
+            let trackingIdentifier = tidTextField.text ?? ""
```
Comment (MathStPierre) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:84]: The method `startTracking` seems to toggle tracking from start to cancel. Maybe we could rename the method `startCancelTracking` or `toggleTracking` 🤔

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist [hunk 7]
```diff
@@ -13,23 +13,45 @@
...
 	<true/>
+	<key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
+	<string>Program requires GPS to track location </string>
 	<key>NSLocationAlwaysUsageDescription</key>
-	<string>UserSessionApp needs Always authorization</string>
-    <key>NSLocationAlwaysAndWhenInUseUsageDescription</key>
-    <string>UserSessionApp needs Always authorization</string>
-    <key>NSLocationWhenInUseUsageDescription</key>
-    <string>UserSessionApp needs Always authorization</string>
+	<string>Program requires GPS to track location </string>
```
Comment (MathStPierre) [inline iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:26]: NSLocationAlwaysUsageDescription is deprecated starting with iOS 11. Maybe we still want to keep it though 🤔 

 https://developer.apple.com/documentation/corelocation

File: .gitignore [hunk 8]
```diff
@@ -1 +1,3 @@
 Curbside.framework
+.DS_Store
+.idea/
```
Comment (radwar13) [inline .gitignore:3]: Why does this have a .idea directory?

File: iOS/UserSessionApp/Swift/Podfile [hunk 9]
```diff
@@ -1,8 +1,8 @@
 # Uncomment the next line to define a global platform for your project
-# platform :ios, '9.0'
+platform :ios, '9.0'
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/Podfile:2]: should change this to 11

File: iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata [hunk 10]
```diff
@@ -2,6 +2,6 @@
 <Workspace
    version = "1.0">
    <FileRef
-      location = "self:/Users/honchen/Development/curbside/ARRIVESDKSampleApps/iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj">
+      location = "self:">
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata:5]: This file should be excluded from the project.

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist (hunk 7 above)
Comment (radwar13) [reply iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:26]: remove

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist [hunk 11]
```diff
@@ -13,23 +13,45 @@
...
+			</array>
+		</dict>
+	</dict>
+	<key>UIApplicationSupportsIndirectInputEvents</key>
+	<true/>
 	<key>UIBackgroundModes</key>
 	<array>
 		<string>fetch</string>
 		<string>location</string>
+		<string>remote-notification</string>
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:54]: why is this added?

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist [hunk 12]
```diff
@@ -13,23 +13,45 @@
...
+		<false/>
+		<key>UISceneConfigurations</key>
+		<dict>
+			<key>UIWindowSceneSessionRoleApplication</key>
+			<array>
+				<dict>
+					<key>UISceneConfigurationName</key>
+					<string>Default Configuration</string>
+					<key>UISceneDelegateClassName</key>
+					<string>$(PRODUCT_MODULE_NAME).SceneDelegate</string>
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:41]: remove the above 2 lines 
```
					<key>UISceneDelegateClassName</key>
 					<string>$(PRODUCT_MODULE_NAME).SceneDelegate</string>

```

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 13]
```diff
@@ -24,27 +24,29 @@ class ViewController: UIViewController, CSUserSessionDelegate {
     @IBOutlet var statusLabel: UILabel!
     @IBOutlet var trackingIdentifierButton: UIButton!
     @IBOutlet var trackingButton: UIButton!
-
     override func viewDidLoad() {
+        
         super.viewDidLoad()
         statusLabel.text = ""
+        resetError()
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:31]: why is there a resetError on viewDidLoad? do you have some defaults in the UI which needs to be cleared out?

Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:31]: The UI on the old app didn't have the error labels cleared out and I just tried to emulate how the old UI was. Will clear out the label and remove this line.

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist (hunk 11 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:54]: not needed - will remove

File: iOS/UserSessionApp/Swift/UserSessionApp/Info.plist (hunk 12 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/Info.plist:41]: When I remove the lines:
```
<key>UISceneDelegateClassName</key>
 <string>$(PRODUCT_MODULE_NAME).SceneDelegate</string>
``` 
the app won't display the UI since a scene delegate class must be specified to use storyboard file.

File: .gitignore (hunk 8 above)
Comment (naylin27) [reply .gitignore:3]: I open it in Intellij by mistake - will remove line since I have removed the file

File: iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata (hunk 10 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/project.xcworkspace/contents.xcworkspacedata]: removed project.xcworkspace directory and added to gitignore

File: iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/xcuserdata/naylin.min.xcuserdatad/xcschemes/xcschememanagement.plist [hunk 14]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp.xcodeproj/xcuserdata/naylin.min.xcuserdatad/xcschemes/xcschememanagement.plist:1]: any user.xcuserdatad file should be ignored from GitHub

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 15]
```diff
@@ -34,17 +34,16 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
         NotificationCenter.default.addObserver(self, selector: #selector(ViewController.sessionValidated(notification:)), name: NSNotification.Name.init(rawValue: kSessionValidatedNotificationName), object: nil)
-        resetError()
         CSUserSession.current().delegate = self
     }
     
     override func viewWillDisappear(_ animated: Bool) {
-        super.viewWillAppear(animated)
-        NotificationCenter.default.removeObserver(self)
-    }
+            super.viewWillAppear(animated)
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:41]: Bad code from original should be super.viewWillDisappear(animated)

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 16]
```diff
@@ -34,17 +34,16 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
+            NotificationCenter.default.removeObserver(self)
+        }
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:43]: why are indents messed up above?

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift [hunk 17]
```diff
@@ -73,49 +72,49 @@ class ViewController: UIViewController, CSUserSessionDelegate {
...
+                }
+                
+                if (siteIdentifier.count == 0) {
+                    errorTitleLabel.text = "Empty site Identifier.";
+                    return;
+                }
+                CSUserSession.current().startTripToSite(withIdentifier: siteIdentifier, trackToken: trackToken)
+            } else {
+                let trackToken = trackTokenField.text
+                CSUserSession.current().cancelTripToSite(withIdentifier: siteIdentifier, trackToken: trackToken)
```
Comment (radwar13) [inline iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:103]: bad code from original. You are stopping the trip not cancelling, you should be calling compelteTrip...

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift (hunk 16 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:43]: fixed

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift (hunk 15 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:41]: fixed

File: iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift (hunk 17 above)
Comment (naylin27) [reply iOS/UserSessionApp/Swift/UserSessionApp/ViewController.swift:103]: fixed

File: .gitignore [hunk 18]
```diff
@@ -1 +1,5 @@
 Curbside.framework
+.DS_Store
+build/
+project.xcworkspace/
+*.xcuserdatad/
```
Comment (amitmats) [inline .gitignore:5]: xcuserdata/

**Comment by radwar13** [comment]
I still don’t see why there is the scene delegate in the app. I dont see it in the current TOT, but the scene delegate has my name on the timestamp from 2017! 
Opening up the diff for project.pbxproj file I see that it has iOS version mismatches I am not sure why those are there
	Podfile has 
		min version of 11
		deployment target of 13
 	project.pbxproj has deployment version of 14.2 

This branch is not on git. This has gone through many to/fros. I have opened the project, did the update and pushed to Swift-5.update branch. Check that out, if it works as expected, merge the branch with master.
